stylix/autoload.nix

```
@@ -42,7 +42,19 @@ builtins.concatLists (
         {
           key = file;
           _file = file;
-          imports = [ (module (args // extraArgs // { inherit mkTarget; })) ];
+          imports = [
+            (module (
+              args
+              // extraArgs
+              // {
+                inherit mkTarget;
+                config = lib.recursiveUpdate config {
+                  stylix = throw "stylix: `config.stylix` accessed while using mkTarget";
+                  lib.stylix.colors = throw "stylix: `config.lib.stylix.colors` accessed while using mkTarget";
```

Following up on the discussion from https://github.com/nix-community/stylix/pull/1368#discussion_r2105059134, should the "`config.stylix` accessed" phrasing be more specific to accessing an "unguarded" or "unrequested" argument:

```suggestion
                  stylix =
                    throw
                    "stylix: unguarded `config.stylix` access inside mkTarget";

                  lib.stylix.colors =
                    throw
                    "stylix: unguarded `config.lib.stylix.colors` access inside mkTarget";
```

The following different structure might be more direct:

```suggestion
                  stylix =
                    throw
                    "stylix: mkTarget accessed unguarded `config.stylix`";

                  lib.stylix.colors =
                    throw
                    "stylix: mkTarget accessed unguarded `config.lib.stylix.colors`";
```

I prefer the second version.